### PR TITLE
changes returned charm name to charmhub url

### PR DESCRIPTION
**Pull request by maxflintoff**
In order to avoid inconsistencies with a charm's self-reported name (from `metadata.yaml`) and the charmhub name this PR changes the name returned from the internal client's `ReadApplication` function to use the CharmURL name instead of the application name.
This change is on line 294

Additionally resolved some linting errors in the `internal/juju/applications.go` file:
L328, L389: ineffectual assignment to err
L268, L271, L282, L290, L326: https://staticcheck.io/docs/checks#S1028
L344: https://staticcheck.io/docs/checks#S1005

### CODE REVIEW

File: internal/juju/applications.go [hunk 1]
```diff
@@ -279,19 +279,19 @@ func (c applicationsClient) ReadApplication(input *ReadApplicationInput) (*ReadA
...
 	// NOTE: we are assuming that this charm comes from CharmHub
 	charmURL, err := charm.ParseURL(appStatus.Charm)
 	if err != nil {
-		return nil, errors.New(fmt.Sprintf("failed to parse charm: %v", err))
+		return nil, fmt.Errorf("failed to parse charm: %v", err)
 	}
 
 	response := &ReadApplicationResponse{
-		Name:     appInfo.Charm,
+		Name:     charmURL.Name,
```
Comment (maxflintoff) [inline internal/juju/applications.go:294]: relevant change for the PR title

File: internal/juju/applications.go [hunk 2]
```diff
@@ -323,7 +323,7 @@ func (c applicationsClient) UpdateApplication(input *UpdateApplicationInput) err
 	var appStatus params.ApplicationStatus
 	var exists bool
 	if appStatus, exists = status.Applications[input.AppName]; !exists {
-		return errors.New(fmt.Sprintf("no status returned for application: %s", input.AppName))
+		return fmt.Errorf("no status returned for application: %s", input.AppName)
```
Comment (maxflintoff) [inline internal/juju/applications.go:326]: https://staticcheck.io/docs/checks#S1028

File: internal/juju/applications.go [hunk 3]
```diff
@@ -380,13 +380,19 @@ func (c applicationsClient) UpdateApplication(input *UpdateApplicationInput) err
 		}
 
 		channel, err := charm.ParseChannel(appStatus.CharmChannel)
```
Comment (maxflintoff) [inline internal/juju/applications.go:382]: ineffectual assignment to err

File: internal/juju/applications.go [hunk 4]
```diff
@@ -380,13 +380,19 @@ func (c applicationsClient) UpdateApplication(input *UpdateApplicationInput) err
...
+		if err != nil {
+			return err
+		}
 
 		origin, err := utils.DeduceOrigin(newURL, channel, platform)
 		if err != nil {
 			return err
 		}
 
 		resultOrigin, err := charmsAPIClient.AddCharm(newURL, origin, false)
```
Comment (maxflintoff) [inline internal/juju/applications.go:392]: ineffectual assignment to err

File: internal/juju/applications.go [hunk 5]
```diff
@@ -341,7 +341,7 @@ func (c applicationsClient) UpdateApplication(input *UpdateApplicationInput) err
 
 		if unitDiff < 0 {
 			var unitNames []string
-			for unitName, _ := range appStatus.Units {
+			for unitName := range appStatus.Units {
```
Comment (maxflintoff) [inline internal/juju/applications.go:344]: https://staticcheck.io/docs/checks#S1005

File: internal/juju/applications.go [hunk 6]
```diff
@@ -279,19 +279,19 @@ func (c applicationsClient) ReadApplication(input *ReadApplicationInput) (*ReadA
...
+		return nil, fmt.Errorf("no status returned for application: %s", input.AppName)
 	}
 
 	unitCount := len(appStatus.Units)
 
 	// NOTE: we are assuming that this charm comes from CharmHub
 	charmURL, err := charm.ParseURL(appStatus.Charm)
 	if err != nil {
-		return nil, errors.New(fmt.Sprintf("failed to parse charm: %v", err))
+		return nil, fmt.Errorf("failed to parse charm: %v", err)
```
Comment (maxflintoff) [inline internal/juju/applications.go:290]: https://staticcheck.io/docs/checks#S1028

File: internal/juju/applications.go [hunk 7]
```diff
@@ -265,10 +265,10 @@ func (c applicationsClient) ReadApplication(input *ReadApplicationInput) (*ReadA
 		return nil, err
 	}
 	if len(apps) > 1 {
-		return nil, errors.New(fmt.Sprintf("more than one result for application: %s", input.AppName))
+		return nil, fmt.Errorf("more than one result for application: %s", input.AppName)
```
Comment (maxflintoff) [inline internal/juju/applications.go:268]: https://staticcheck.io/docs/checks#S1028

**Review by night0wl — APPROVED** [review]
LGTM 👍

**Pull request merged**